js/pop-up.js

```diff
@@ -0,0 +1,230 @@
+    imagealt: 'Screenshot of Sign-up Form Project',
+    description:
+      'A daily selection of privately personalized reads; no accounts or sign-ups required.',
+    fulldescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s. Lorem Ipsum is simply dummy text of the printing and typesetting industry.
+      
+      Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum han printer took a galley of type and scrambled it 1960s with the releawn printer took a galley of type and scrambled it 1960s.`,
+    framework: ['html', 'css', 'javascript'],
+    frameworks: ['html', 'css', 'javascript', 'Ruby', 'BootStrap'],
+    btnId: 'art-work-btn-element',
+    seelivelink: `https://deploy-preview-1--add-top-signup-form.netlify.app/`,
+    seesourcelink: `https://github.com/moshoodabdullahi/top-signup-form/pull/1`,
+  },
+];
+
+const generateDetailsWork = (
+  id,
+  title,
+  tags,
+  image,
+  imagealt,
+  fulldescription,
+  framework,
+  frameworks,
+  seelivelink,
+  seesourcelink,
+) =>
+  `<article id="${id}" class="details-target-element details-target-element details-works-article details-popup left">
+<div class="works-article details-work-article d-block">
+  <div class="details-header">
+    <h3 class="works-headline details-works-headline">
+      ${title}
+      <svg id="close-details-element" class="close-details" width="30" height="30" viewBox="0 0 25 29" fill="none"
+        xmlns="http://www.w3.org/2000/svg">
+        <path fill-rule="evenodd" clip-rule="evenodd"
+          d="M5.29289 5.29289C5.68342 4.90237 6.31658 4.90237 6.70711 5.29289L12 10.5858L17.2929 5.29289C17.6834 4.90237 18.3166 4.90237 18.7071 5.29289C19.0976 5.68342 19.0976 6.31658 18.7071 6.70711L13.4142 12L18.7071 17.2929C19.0976 17.6834 19.0976 18.3166 18.7071 18.7071C18.3166 19.0976 17.6834 19.0976 17.2929 18.7071L12 13.4142L6.70711 18.7071C6.31658 19.0976 5.68342 19.0976 5.29289 18.7071C4.90237 18.3166 4.90237 17.6834 5.29289 17.2929L10.5858 12L5.29289 6.70711C4.90237 6.31658 4.90237 5.68342 5.29289 5.29289Z"
+          fill="black" />
+      </svg>
+    </h3>
+  </div>
+  <ul class="details-about-icon d-flex">
+  ${tags
+    .map((tag) => '<li class="details-content">' + tag + '</li>')
+    .join('<li class="dot details-dot"></li>')}
+  </ul>
+
+  <div class="img-profile details-img-profile">
+    <img class="details-works-img" src="${image}" alt="${imagealt}" />
+  </div>
+
+  <p class="works-writeup details-works-writeup d-block-lg">
+  ${fulldescription}
+  </p>
+
+  <ul class="abt-link details-abt-link d-flex">
+  ${frameworks
+    .map((framework, index) => {
```

Comment:
please i don't seem to grasp this aspect of your code ,mind explaining it to me 🙃🙃